src/rpc/util.h

```diff
@@ -335,12 +335,48 @@ struct RPCExamples {
     std::string ToDescriptionString() const;
 };
 
+template <typename T>
+struct RPCParam
+{
+    const RPCArg& m_arg;
+    const UniValue& m_value;
+    std::optional<T> m_defaults = {};
+    RPCParam<T>& defaults(const T& defaults) {
+        CHECK_NONFATAL(m_arg.m_fallback.index() != 2);
+        m_defaults = defaults;
+        return *this;
+    }
```

Comment:
I don't understand how any of this works. I removed this and everything compiled just fine:

```diff
diff --git a/src/rpc/util.h b/src/rpc/util.h
index f7a1acd1b5..1d5b7fc318 100644
--- a/src/rpc/util.h
+++ b/src/rpc/util.h
@@ -348,11 +348,6 @@ struct RPCParam
     const RPCArg& m_arg;
     const UniValue& m_value;
     std::optional<T> m_defaults = {};
-    RPCParam<T>& defaults(const T& defaults) {
-        CHECK_NONFATAL(m_arg.m_fallback.index() != 2);
-        m_defaults = defaults;
-        return *this;
-    }
     T get() const {
         if (!m_value.isNull()) return value(m_value);
         if (m_defaults) return *m_defaults;
```

I think it would make sense to split this into at least two commits:
* First one to introduce `RPCContext`
* Second one to introduce `RPCParam`

Finally, would be nice to use clang-format on new code.

Reply:
https://github.com/bitcoin/bitcoin/pull/20017/files#r682710976

@MarcoFalke yup, not used in this branch.

Reply:
@promag are you still working on this?

Reply:
I’ll rebase and remove dead code.